filters/auth/oauth.go

```diff
@@ -159,7 +162,14 @@ func jsonGet(url *url.URL, auth string, doc interface{}) error {
 		return err
 	}
 
-	rsp, err := http.DefaultClient.Do(req)
+	var client *http.Client
+	if timeout != 0 {
+		client = &http.Client{Timeout: timeout}
```

Comment:
This will not failover DNS if someone will change the traffic to another loadbalancer for example.
We need to have the workaround as in the kubernetes dataclient or the proxy.
See https://github.com/zalando/skipper/blob/master/dataclients/kubernetes/kube.go#L310-L337.

Reply:
@szuecs Is this still true since the introduction of Transport.IdleConnTimeout in go 1.7? I think that new setting which defaults to 90secs is good enough to mitigate that problem.

Reply:
@szuecs do I need to create the same Transport dialer in the auth filter?

Reply:
@addityasingh you don't need the same, but you need the `for {}` loop below the transport to close idle connections to make the http client to do a DNS call to make DNS based failover happen.